lua/ibl/highlights.lua

```diff
@@ -15,7 +15,16 @@ local M = {
 }
 
 local get = function(name)
-    return vim.api.nvim_get_hl(0, { name = name })
+    local hl
+    if vim.api.nvim_get_hl then -- check for new neovim 0.9 API
+        hl = vim.api.nvim_get_hl(0, { name = name })
+    else
+        hl = vim.api.nvim_get_hl_by_name(name, vim.o.termguicolors)
+        hl.fg, hl.bg = hl.foreground, hl.background
+        hl.ctermfg, hl.ctermbg = hl.fg, hl.bg
+        hl.sp = hl.special
```

Comment:
this only covers the colors, but `nvim_get_hl` returns much more. It needs to cover all of these fields.

- fg (or foreground): color name or "#RRGGBB", see note.
- bg (or background): color name or "#RRGGBB", see note.
- sp (or special): color name or "#RRGGBB"
- blend: integer between 0 and 100
- bold: boolean
- standout: boolean
- underline: boolean
- undercurl: boolean
- underdouble: boolean
- underdotted: boolean
- underdashed: boolean
- strikethrough: boolean
- italic: boolean
- reverse: boolean
- nocombine: boolean
- link: name of another highlight group to link to, see |:hi-link|.
- default: Don't override existing definition |:hi-default|
- ctermfg: Sets foreground of cterm color |ctermfg|
- ctermbg: Sets background of cterm color |ctermbg|
- cterm: cterm attribute map, like |highlight-args|. If not set, cterm attributes will match those from the attribute map documented above.
- force: if true force update the highlight group when it exists.

Reply:
This function still carries all of those things. I removed the unnecessary renaming here.